api/api_hierarchical.go

```diff
@@ -17,7 +17,7 @@ type HierarchicalCns interface {
 	AddSubnet(ctx context.Context, wallet address.Address, parent address.SubnetID, name string, cns uint64, minerStake abi.TokenAmount, checkPeriod abi.ChainEpoch, p *hierarchical.ConsensusParams) (address.Address, error) // perm:write
 	JoinSubnet(ctx context.Context, wallet address.Address, value abi.TokenAmount, id address.SubnetID, validatorAddr string) (cid.Cid, error)                                                                                 // perm:write
 	SyncSubnet(ctx context.Context, id address.SubnetID, stop bool) error                                                                                                                                                      // perm:write
-	MineSubnet(ctx context.Context, wallet address.Address, id address.SubnetID, stop bool) error                                                                                                                              // perm:read
+	MineSubnet(ctx context.Context, wallet address.Address, id address.SubnetID, stop bool, p *hierarchical.MiningParams) error                                                                                                // perm:read
```

Comment:
Now that I see the permissions of this function, maybe we should change them to `perm:write`. What do you think?

Reply:
It depends on what `perm:write` means. We don't change the system state. Right? We outputted log and before that... but that was another place.

Reply:
Yes, my question is not with your change, but that when reading it I just realized that maybe that mine should have write instead of read permissions. Is not critical.